en/net/advanced-operations/securing-and-signing/digitally-sign-pdf-file/_index.md

```diff
@@ -279,30 +318,38 @@ The example below creates a non-detached signature with the RSA algorithm and th
 If you use PKCS7Detached instead of PKCS7, you can use ECDCA and set the desired digest algorithm.
 
 ```csharp
-var inputPdf = "111.pdf";
-var inputP12 = "111.p12";
-var inputPfxPassword = "123456";
-var outputPdf = "111_out.pdf";
-using (var sign = new PdfFileSignature())
+// For complete examples and data files, check for https://github.com/aspose-pdf/Aspose.PDF-for-.NET
+  
+private static void SignWithCertificate(string pfxFilePath, string password)
 {
-    sign.BindPdf(inputPdf);
-    var pkcs7 = new PKCS7(inputP12, inputPfxPassword);// You can use PKCS7Detached with digest algorithm argument.
-    pkcs7.CustomSignHash = CustomSignHash;
-    sign.Sign(1, "reason", "cont", "loc", false, new System.Drawing.Rectangle(0, 0, 500, 500), pkcs7);
-    sign.Save(outputPdf);
-}
+    // The path to the documents directory
+    string dataDir = RunExamples.GetDataDir_AsposePdf_SecuritySignatures();
+    
+    using (var sign = new Aspose.Pdf.Facades.PdfFileSignature())
+    {   
+        sign.BindPdf(dataDir + "input.pdf");
+        // Create PKCS#7 object to sign
+        var pkcs7 = new Aspose.Pdf.Forms.PKCS7(string pfxFilePath, string password);// You can use PKCS7Detached with digest algorithm argument
+        // Set the delegate to external sign
+        pkcs7.CustomSignHash = CustomSignHash;
+        // Sign the file
+        sign.Sign(1, "reason", "cont", "loc", false, new System.Drawing.Rectangle(0, 0, 500, 500), pkcs7);
+        // Save the document
+        sign.Save(dataDir + "SignWithCertificate_out.pdf");
+    }
 
-// Custom hash signing function to generate a digital signature
-private byte[] CustomSignHash(byte[] signableHash, DigestHashAlgorithm digestHashAlgorithm)
-{
-    var inputP12 = "111.p12";
-    var inputPfxPassword = "123456";
-    X509Certificate2 signerCert = new X509Certificate2(inputP12, inputPfxPassword, X509KeyStorageFlags.Exportable);
-    RSACryptoServiceProvider rsaCSP = new RSACryptoServiceProvider();
-    var xmlString = signerCert.PrivateKey.ToXmlString(true);
-    rsaCSP.FromXmlString(xmlString);
-    byte[] signedData = rsaCSP.SignData(signableHash, CryptoConfig.MapNameToOID("SHA1"));
-    return signedData;
+    // Custom hash signing function to generate a digital signature
+    byte[] CustomSignHash(byte[] signableHash, DigestHashAlgorithm digestHashAlgorithm)
+    {
+        var inputP12 = "111.p12";
+        var inputPfxPassword = "123456";
+        X509Certificate2 signerCert = new X509Certificate2(inputP12, inputPfxPassword, X509KeyStorageFlags.Exportable);
+        RSACryptoServiceProvider rsaCSP = new RSACryptoServiceProvider();
```

Comment:
We may not specify full qualified name for the System libraries. However, `System.Security.Cryptography` is enough hard case to specify it.
Additionally `var` can be used.